### PR TITLE
Script for defect 2442 json lib

### PR DESCRIPTION
**Pull request by TMyza**
ATF Test Scripts to check [#2442](https://github.com/smartdevicelink/sdl_core/issues/2442)

Summary
A set of test scripts to verify that issue is fixed

This PR is **ready** for review.

### CODE REVIEW

File: test_scripts/Defects/5_0/2442_Fix_mem_seg_fault_error_in_jsoncpp.lua [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+-- User story: https://github.com/smartdevicelink/sdl_core/issues/2442
+--
+-- Description:
+-- The root cause of the issue is a stack overflow which is occurring due to recursive call of deeply nested areas 
+-- and objects. We have readValue(..) which calls readObject(..) or readArray(..), which call readValue(...). 
+-- Steps to reproduce:
+-- 1) Generate Json file with more than 50 nestings.
+-- 2) The starting point is a call to parse(..).
+-- Expected:
+-- 1) When recursion threshold is exceeded (50 nestings) SDL should STOP.
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2442_Fix_mem_seg_fault_error_in_jsoncpp.lua]: SDL must not stop but `When recursion threshold is
exceeded we return invalid messege`

File: test_scripts/Defects/5_0/2442_Fix_mem_seg_fault_error_in_jsoncpp.lua [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    test:FailTestCase(pCause)
+end
+
+local  function checkStatusSDL()    
+    if SDL:CheckStatusSDL() == SDL.STOP  then        
+        return true, "SDL was stoped"
+    end
+    if SDL:CheckStatusSDL() == SDL.RUNNING then
+        failTestCase("SDL was not stopped")
+    end
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2442_Fix_mem_seg_fault_error_in_jsoncpp.lua]: Please do not forget about processing SDL.CRASH status

File: test_scripts/Defects/5_0/2442_Fix_mem_seg_fault_error_in_jsoncpp.lua [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ --[[ Scenario ]]
+ runner.Title("Preconditions")
+ runner.Step("Backup HMI capabilities file", commonRC.backupHMICapabilities)
+ runner.Step("Clean environment", commonRC.preconditions)
+
+ -- [[ Test ]]
+runner.Title("Test")
+runner.Step("Start SDL, HMI, connect Mobile, start Session", funcCausesStopOfSDL)
+runner.Step("Check status SDL", checkStatusSDL)
+
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2442_Fix_mem_seg_fault_error_in_jsoncpp.lua]: add also case with json from mobile

**Comment by dboltovskyi** [comment]
Livio team please do not review this PR for now since it hasn't passed our internal review yet.  
Review and appropriate updates going be done till 1st of November.
Thank you.

**Comment by iCollin** [comment]
@TMyza @dboltovskyi @GetmanetsIrina 

I see that PR is marked as a draft and not ready, but script seems relevant and does test the issue.

Would you still like this to be merged?

**Comment by dboltovskyi** [comment]
@iCollin Yes, we would like this script to be merged.
Please notice:
1. Description is corrected: `not ready` => `ready`
2. Script moved from `5_0` to `6_1` folder in 3498f51

**Pull request merged**